### PR TITLE
3.0.4: Improved request-response example

### LINKED ISSUE #1720: Inconsistent usage of quotes in yaml examples
This is a minor nitpick but as someone unfamiliar with the OpenAPI specifcation and yaml until now, I found it confusing at first.  There is inconsistent usage of quotes in the yaml examples in the [3.0.2 specification](https://github.com/OAI/OpenAPI-Specification/blob/3.0.2/versions/3.0.2.md) (and presumably earlier).

Specifically, there are places where the yaml examples have content `'application/json'` and other spots where it has `application/json`.  Same confusion applies to other content values.  A notable example is the [file uploads section](https://github.com/OAI/OpenAPI-Specification/blob/3.0.2/versions/3.0.2.md#considerations-for-file-uploads) where it has content `application/octet-stream` without quotes, `'image/jpeg'` and `'image/png'` with single quotes, and then `multipart/form-data` without quotes.

I also noticed inconsistencies with the `$ref: value` where for example it has `$ref: "#/components/schemas/Pet"` with double quotes and `$ref: '#/components/schemas/SomePayload'` with single quotes.

Is there a best practice?  Reading the examples it wasn't immediately clear to me when I should quote vs not quote, and if single quote vs double quote is more standard.

### PR DESCRIPTION
**Pull request by ralfhandl**
* separate code blocks for request and response message
* adjusted content-length to what my editor claims it to be

Fixes #1720

### CODE REVIEW

File: versions/3.0.4.md [hunk 1]
```diff
@@ -2036,17 +2036,21 @@ For example, given the following HTTP request:
 POST /subscribe/myevent?queryUrl=https://clientdomain.com/stillrunning HTTP/1.1
 Host: example.org
 Content-Type: application/json
-Content-Length: 187
+Content-Length: 188
```
Comment (handrews) [inline versions/3.0.4.md:2039]: I'm confused by both numbers- to me it looks like it's 260 if there is a trailing newline (more if the line ends are CRLF):

```PYCON
Python 3.8.13 (default, May 11 2022, 14:42:53) 
[Clang 13.1.6 (clang-1316.0.21.2.3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> x = """{
...   "failedUrl" : "https://clientdomain.com/failed",
...   "successUrls" : [
...   "failedUrl": "https://clientdomain.com/failed",
...   "successUrls": [
...     "https://clientdomain.com/fast",
...     "https://clientdomain.com/medium",
...     "https://clientdomain.com/slow"
...   ]
... }
... """
>>> x
'{\n  "failedUrl" : "https://clientdomain.com/failed",\n  "successUrls" : [\n  "failedUrl": "https://clientdomain.com/failed",\n  "successUrls": [\n    "https://clientdomain.com/fast",\n    "https://clientdomain.com/medium",\n    "https://clientdomain.com/slow"\n  ]\n}\n'
>>> len(x)
260
>>> 
```

**Review by handrews — COMMENTED** [review]
I'm pretty sure I'm missing something here with the content-length, but I'm not sure what?

File: versions/3.0.4.md (hunk 1 above)
Comment (ralfhandl) [reply versions/3.0.4.md:2039]: Funny, I did something similar with Node.js and get 188:
~~~sh
Welcome to Node.js v22.5.1.
Type ".help" for more information.
> var text = `{
...   "failedUrl": "https://clientdomain.com/failed",
...   "successUrls": [
...     "https://clientdomain.com/fast",
...     "https://clientdomain.com/medium",
...     "https://clientdomain.com/slow"
...   ]
... }`
undefined
> text
'{\n' +
  '  "failedUrl": "https://clientdomain.com/failed",\n' +
  '  "successUrls": [\n' +
  '    "https://clientdomain.com/fast",\n' +
  '    "https://clientdomain.com/medium",\n' +
  '    "https://clientdomain.com/slow"\n' +
  '  ]\n' +
  '}'
> text.length
188
~~~

Which is identical to what my editor shows when I select that text block in the markdown source.

Comment (ralfhandl) [reply versions/3.0.4.md:2039]: Out of curiosity I repeated the experiment with Python:
```sh
Python 3.9.6 (default, Feb  3 2024, 15:58:27) 
[Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> text="""{
...   "failedUrl": "https://clientdomain.com/failed",
...   "successUrls": [
...     "https://clientdomain.com/fast",
...     "https://clientdomain.com/medium",
...     "https://clientdomain.com/slow"
...   ]
... }"""
>>> text
'{\n  "failedUrl": "https://clientdomain.com/failed",\n  "successUrls": [\n    "https://clientdomain.com/fast",\n    "https://clientdomain.com/medium",\n    "https://clientdomain.com/slow"\n  ]\n}'
>>> len(text)
188
```
Which produces the same length of 188 as Node.js.

Comment (ralfhandl) [reply versions/3.0.4.md:2039]: Now I see it: your experiment contains duplicated lines:
image

Comment (mikekistler) [reply versions/3.0.4.md:2039]: I reproduced Ralf's experiment with the same result, so 188 does seem to be the right length. But I think the result of `len(text)` in Python is in units of unicode characters, which may not be the length of a UTF-8-encoded response body (though in this case it is). The way to get the length of a string encoded to UTF-8 in Python is

```python
>>> len(text.encode('utf-8'))
188
```

Comment (handrews) [reply versions/3.0.4.md:2039]: 188 it is!  I swear I stared at that and could not figure out what was going on.... sorry to take up your time, folks!

**Review by handrews — APPROVED** [review]

**Review by mikekistler — APPROVED** [review]
Looks good! 👍

**Pull request merged**